components/utils/src/links.rs

```diff
@@ -0,0 +1,16 @@
+pub fn anchor_id_checks(anchor:&str) -> Vec<String> {
+    vec![
+        format!(" id={}", anchor),
```

Comment:
Will regex be slower or faster here?

Reply:
Not sure, this is a lot of allocation, i'm guessing the regex would be faster? But hard to say without a benchmark

Reply:
Possibly faster and probably more maintainable. I just copied what was already there, but happy to refactor it out to use a regexp. This code only runs if other anchor checks fail, so any performance hit will only be on sites that are broken or have these anchors in.

The regexp crate is already a dependency, so nothing wrong with trying. Am guessing we'd need a few hundred links to make a benchmark informative.

Reply:
let's go with the regex approach then

Reply:
Okay, pushed. Performance testing results.

I compiled site with a two small files but one contain ~2k links to an XML anchor. I then tested various options, in all cases using `cargo run` but taking the time reported at the end.

Comparing regexp to direct string matching

```
regexp:
   --release: 1.3s
   no --release: 39s

non-regexp:
   --release: 1.0s
   no --release: 19.4s
```

Conclusion: Regexp is slower esp for non-optimized builds.

Then comparing the non-regexp implementation with heuristic_link_check false or true

```
heuristic_link_check = false --release   0.5s 
heuristic_link_check = false no --release 1.1s
```
Conclusion: The  heuristic link check causes a 100% in build times (for release build) and a much bigger increase for non-optimized.

However, in this case zola produces lots of error messages and printouts can be slow. So, I have also tried fixing the broken links by adding an markdown generated anchor (with the same ~2k links to it). I also tried comparing this situation with and without the heuristic link checker.

```
--release, no failures: 0.5s
no -- release, no failures: 1.1s

heuristic_link_check = false --release no failures 0.5s
heuristic_link_check = false no --release no failures 1.1s
```

Conclusion 1: printing errors doesn't make much of a difference to the performance.
Conclusion 2: if the heuristic link checker is on, it does not make that much difference if most links are not broken. This would be predicted from the code.

So, I would say the regexp code is a bit neater, the multiple format! checking is a bit faster, so either implementation is fine. I'd probably got with the regexp cause neat trumps fast for me, but happy with what ever you prefer.

And while the heuristic_link_checker is significantly slows things down, you only pay for it if you are using XML anchors. As zola current refuses to build a site under these circumstances, it seems a reasonable price to pay.